src/rgw/rgw_gc.cc

```diff
@@ -464,7 +625,7 @@ int RGWGC::process(int index, int max_secs, bool expired_only,
 	} // chains loop
       } // else -- chains not empty
     } // entries loop
-    if (entries.size() > 0) {
+    if (transitioned_objects_cache[index] && entries.size() > 0) {
```

Comment:
@cbodley : should the queue entries be removed once the tail objects have been actually deleted, something similar to schedule_tag_removal()? in batches of trim_chunk_size() ?